ludwig/hyperopt/sampling.py

```diff
@@ -282,7 +295,11 @@ def sample(self) -> Dict[str, Any]:
     def update(self, sampled_parameters: Dict[str, Any], metric_score: float):
         for key in sampled_parameters:
             if key in self.cat_params_values_types:
-                sampled_parameters[key] = str(sampled_parameters[key])
+                if type(sampled_parameters[key]) == list:
```

Comment:
Maybe here to be more consistent we can say `is not in {bool, int, float, str}`? This will make it work also for dicts and other json-serializable types.

Reply:
good point. will make the necessary edits.